### PR TITLE
Mongodb options

### PR DESCRIPTION
**Pull request by brasseld**
Extends MongoDB configuration and fix some other options

### CODE REVIEW

File: apim/templates/gateway-configmap.yaml [hunk 1]
```diff
@@ -34,11 +34,19 @@ data:
...
+        {{- if .Values.mongo.uri }}
+        uri: {{ .Values.mongo.uri }}
+        {{ else }}
+        uri: mongodb://{{- if (eq .Values.mongo.auth.enabled true) }}{{ .Values.mongo.auth.username }}:{{ .Values.mongo.auth.password }}@{{- end }}{{ .Values.mongo.dbhost }}:{{ .Values.mongo.dbport }}/{{ .Values.mongo.dbname }}?{{- if .Values.mongo.rsEnabled }}&replicaSet={{ .Values.mongo.rs }}{{- end }}{{- if (eq .Values.mongo.auth.enabled true) }}&authSource={{ .Values.mongo.auth.source }}{{- end }}{{- if .Values.mongo.connectTimeoutMS }}&connectTimeoutMS={{ .Values.mongo.connectTimeoutMS }}{{- end }}
+        {{- end }}
     ratelimit:
       type: mongodb
       mongodb:
-        uri: mongodb://{{- if (eq .Values.mongo.auth.enabled true) }}{{ .Values.mongo.auth.username }}:{{ .Values.mongo.auth.password }}@{{- end }}{{ .Values.mongo.dbhost }}:{{ .Values.mongo.dbport }}/{{ .Values.mongo.dbname }}{{- if .Values.mongo.rsEnabled }}?replicaSet={{ .Values.mongo.rs }}{{- end }}{{- if (eq .Values.mongo.auth.enabled true) }}&authSource=admin{{- end }}{{- if .Values.mongo.connectTimeoutMS }}&connectTimeoutMS={{ .Values.mongo.connectTimeoutMS }}{{- end }}
+        {{- if .Values.mongo.uri }}
```
Comment (chicco785) [inline apim/templates/gateway-configmap.yaml]: then this should be documented and added in values.yaml

File: apim/templates/gateway-configmap.yaml [hunk 2]
```diff
@@ -34,11 +34,19 @@ data:
     management:
       type: mongodb
       mongodb:
-        uri: mongodb://{{- if (eq .Values.mongo.auth.enabled true) }}{{ .Values.mongo.auth.username }}:{{ .Values.mongo.auth.password }}@{{- end }}{{ .Values.mongo.dbhost }}:{{ .Values.mongo.dbport }}/{{ .Values.mongo.dbname }}{{- if .Values.mongo.rsEnabled }}?replicaSet={{ .Values.mongo.rs }}{{- end }}{{- if (eq .Values.mongo.auth.enabled true) }}&authSource=admin{{- end }}{{- if .Values.mongo.connectTimeoutMS }}&connectTimeoutMS={{ .Values.mongo.connectTimeoutMS }}{{- end }}
+        {{- if .Values.mongo.uri }}
+        uri: {{ .Values.mongo.uri }}
```
Comment (chicco785) [inline apim/templates/gateway-configmap.yaml]: then this should be documented and added in values.yaml

File: apim/templates/gateway-configmap.yaml (hunk 1 above)
Comment (brasseld) [reply apim/templates/gateway-configmap.yaml]: Fixed

**Comment by chicco785** [comment]
LGTM

**Pull request merged**